xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -183,8 +183,27 @@ bool CGUIDialogVideoInfo::OnMessage(CGUIMessage& message)
       }
       else if (iControl == CONTROL_BTN_DIRECTOR)
       {
-        std::string strDirector = StringUtils::Join(m_movieItem->GetVideoInfoTag()->m_director, g_advancedSettings.m_videoItemSeparator);
-        OnSearch(strDirector);
+        auto directors = m_movieItem->GetVideoInfoTag()->m_director;
+        if (directors.size() > 1)
+        {
+          auto pDlgSelect = static_cast<CGUIDialogSelect*>(g_windowManager.GetWindow(WINDOW_DIALOG_SELECT));
+          pDlgSelect->Reset();
+          pDlgSelect->SetHeading(CVariant{22080});
+          for (const auto &director: directors)
+            pDlgSelect->Add(director);
+          pDlgSelect->Open();
+
+          auto iItem = pDlgSelect->GetSelectedItem();
+          if (iItem < 0)
+            return true;
+          OnSearch(directors[iItem]);
+
+        }
+        else
+        {
+          auto strDirector = StringUtils::Join(m_movieItem->GetVideoInfoTag()->m_director, g_advancedSettings.m_videoItemSeparator);
```

Comment:
Do we need to join in case there is only one director? Should we change the check to

```
if (m_director.size() == 0)
  return true;
else if (m_director.size() == 1)
  OnSearch(m_director[0])
else if (m_director.size() > 1)
  // select dialog code
```